File: M/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/main.go
package main

import (
	"fmt"
	"sort"
)

func maxArea(h int, w int, horizontalCuts []int, verticalCuts []int) int {
	horizontalCuts = append(horizontalCuts, h)
	verticalCuts = append(verticalCuts, w)
	sort.Ints(horizontalCuts)
	sort.Ints(verticalCuts)
	bh, bw := 0, 0
	hmax, wmax := 0, 0
	for i := 0; i < len(horizontalCuts); i++ {
		if horizontalCuts[i]-bh > hmax {
			hmax = horizontalCuts[i] - bh
		}
		bh = horizontalCuts[i]
	}

	for i := 0; i < len(verticalCuts); i++ {
		if verticalCuts[i]-bw > wmax {
			wmax = verticalCuts[i] - bw
		}
		bw = verticalCuts[i]
	}
	return (wmax * hmax) % 1000000007
}

func main() {
	fmt.Println("a")
}

File: M/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/main.rs
struct Solution {}

impl Solution {
  pub fn max_area(h: i32, w: i32, horizontal_cuts: Vec<i32>, vertical_cuts: Vec<i32>) -> i32 {
    let mut hc = horizontal_cuts;
    hc.push(h);
    hc.push(0);
    let mut vc = vertical_cuts;
    vc.push(w);
    vc.push(0);
    hc.sort();
    vc.sort();

    let mut max_h = 0;
    let mut max_v = 0;
    for i in 1..hc.len() {
      if hc[i] - hc[i - 1] > max_h {
        max_h = hc[i] - hc[i - 1];
      }
    }
    for i in 1..vc.len() {
      if vc[i] - vc[i - 1] > max_v {
        max_v = vc[i] - vc[i - 1];
      }
    }

    (((max_h as i64) * (max_v as i64)) % 1000000007) as i32
  }
}
